src/database/import.ts

```diff
@@ -190,3 +213,63 @@ export default class DatabaseImporter {
     return [root, key].join("/").replace("//", "/");
   }
 }
+
+/**
+ * Batches chunked JSON data up to the specified byte size limit. The emitted batch is an object
+ * containing chunks keyed by pathname.
+ */
+class BatchChunks extends stream.Transform {
+  private batch: Data = { json: {}, pathname: "" };
+  private size = 0;
+
+  constructor(private maxSize: number, opts?: stream.TransformOptions) {
+    super({ ...opts, objectMode: true });
+  }
+
+  _transform(chunk: Data, _: BufferEncoding, callback: stream.TransformCallback) {
+    const chunkSize = JSON.stringify(chunk.json).length + chunk.pathname.length;
```

Comment:
Don't love this extra call to `JSON.stringify`, should probably make it part of `Data`?

Reply:
Ok, I got it to count while chunking.